Add doc for `APPROX_COUNT_DISTINCT` aggregate function (#20188)
pingcap/docs: functions-and-operators/aggregate-group-by-functions.md

    1 row in set (0.00 sec)
    ```

+ `APPROX_COUNT_DISTINCT(expr, [expr...])`

    This function is similar to `COUNT(DISTINCT)` in counting the number of distinct values but returns an approximate result. It uses the `BJKST` algorithm, significantly reducing memory consumption when processing large datasets with a power-law distribution. Moreover, for low-cardinality data, this function provides high accuracy while maintaining efficient CPU utilization.

    The following example shows how to use this function:

    ```sql
    DROP TABLE IF EXISTS t;
    CREATE TABLE t(a INT, b INT, c INT);
    INSERT INTO t VALUES(1, 1, 1), (2, 1, 1), (2, 2, 1), (3, 1, 1), (5, 1, 2), (5, 1, 2), (6, 1, 2), (7, 1, 2);
    ```

    ```sql
    SELECT APPROX_COUNT_DISTINCT(a, b) FROM t GROUP BY c;
    ```

    ```
    +-----------------------------+
    | approx_count_distinct(a, b) |
    +-----------------------------+
    |                           3 |
    |                           4 |
    +-----------------------------+
    2 rows in set (0.00 sec)
    ```

> Except for the `GROUP_CONCAT()`, `APPROX_PERCENTILE()` and `APPROX_COUNT_DISTINCT` functions, all the preceding functions can serve as [Window functions](/functions-and-operators/window-functions.md).

Reviewer: ```suggestion
Except for the `GROUP_CONCAT()`, `APPROX_PERCENTILE()`, and `APPROX_COUNT_DISTINCT` functions, all the preceding functions can serve as [Window functions](/functions-and-operators/window-functions.md).
```

Except for the `GROUP_CONCAT()` and `APPROX_PERCENTILE()` functions, all the preceding functions can serve as [Window functions](/functions-and-operators/window-functions.md).

> + `APPROX_COUNT_DISTINCT(expr)`

Reviewer 2: `APPROX_COUNT_DISTINCT(expr,[expr...])`

Author: > `APPROX_COUNT_DISTINCT(expr,[expr...])`

done

>     This function returns the approximate distinct count of `expr`. It uses `BJKST` algorithm and consumes less memory during simultaneous computation of cardinality for a large number of data sets whose cardinality has power law distribution. Moreover, it's very accurate for data sets with small cardinality and very efficient on CPU.

Reviewer 2: Should we explain that its usage is almost same like `COUNT(DISTINCT)` ?

Author: > Should we explain that its usage is almost same like `COUNT(DISTINCT)` ?

done

>     The usage of this function is almost same with `COUNT(DISTINCT)` but returns approximate result. It uses `BJKST` algorithm and consumes less memory during simultaneous computation of cardinality for a large number of data sets whose cardinality has power law distribution. Moreover, it's very accurate for data sets with small cardinality and very efficient on CPU.

Reviewer: ```suggestion
    This function is similar to `COUNT(DISTINCT)` in counting the number of distinct values but returns an approximate result. It uses the `BJKST` algorithm, significantly reducing memory consumption when processing large datasets with a power-law distribution. Moreover, for low-cardinality data, this function provides high accuracy while maintaining efficient CPU utilization.
```

    The following example shows how to use this function:

    ```sql
    DROP TABLE IF EXISTS t;
    CREATE TABLE t(a INT, b INT);
    INSERT INTO t VALUES(1, 1), (2, 1), (2, 1), (3, 1), (5, 2), (5, 2), (6, 2), (7, 2);
    ```

    ```sql

>     SELECT APPROX_COUNT_DISTINCT(a) FROM t GROUP BY b;

Reviewer 2: Shall we use multiple columns in this expr as an example ?

Author: > Shall we use multiple columns in this expr as an example ?

added

    ```

>     ```sql

Reviewer: ````suggestion
    ```sql
````

>     ```sql

Author: ```suggestion
    ```
```
